src/server.js

```diff
@@ -39,23 +47,25 @@ class ExecSession {
     this.execStream = await this.exec.start(this.attachOptions);
 
     //handling output
-    this.strout = through((data) => {
+    this.strout = through2((data, enc, cb) => {
       this.sendMessage(msgcode.stdout, new Buffer(data));
+      cb();
     });
 
-    this.strerr = through((data) => {
+    this.strerr = through2((data, enc, cb) => {
       this.sendMessage(msgcode.stderr, new Buffer(data));
+      cb();
     });
```

Comment:
I think we're doing this wrong...
We should probably do:
```js
this.stderr = through2((data, enc, cb) => {
  cb(null, Buffer.concat([new Buffer([msgcode.stderr]), data]))
});
this.stderr.pipe(this.strbuf);
this.stdout.pipe(this.strbuf);

this.strbuf.on('data', function(data) {
  outstandingBytes += data.length;
  if (outstandingBytes > MAX_OUTSTANDING_BYTES) {
    this.strbuf.pause();
  }
  this.socket.send(data, {binary: true}, () => {
    outstandingBytes -= data.length;
    this.strbuf.resume();
  });
});
this.strbuf.resume();
```

This way we really only do transformations inside the streams... Transforms in the sense that we add the right `msgcode`, and we get resume/pause for free with `.pipe`. However, we might need to check if `exec.modem.demuxStream` actually pauses when output streams we give it tell it that they are full.